codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithypython/common/nameresolver/SmithyNameResolver.java

```diff
@@ -0,0 +1,339 @@
+      String smithyNamespace, GenerationContext codegenContext) {
+    return getPythonModuleSmithygeneratedPathForSmithyNamespace(
+        smithyNamespace, codegenContext.settings());
+  }
+
+  /**
+   * Given the namespace of a Smithy shape, returns a Pythonic access path to the namespace that can
+   * be used to import shapes from its Smithy-generated namespace.
+   *
+   * @param smithyNamespace
+   * @param settings
+   * @return
+   */
+  public static String getPythonModuleSmithygeneratedPathForSmithyNamespace(
+      String smithyNamespace, PythonSettings settings) {
+    String pythonModuleName;
+    String namespace;
+    // `smithy.api.Unit:`
+    // Smithy-Dafny will generate a stand-in shape in the service
+    if ("smithy.api".equals(smithyNamespace)) {
+      // In the case of a wrappedLocalService shim in a different namespace,
+      // the default modulename should not be used,
+      // and we need a mechanism to override the default modulename.
+      // If the smithy.api namespace has a dependency-module-name mapping, use that.
+      try {
+        pythonModuleName = getPythonModuleNameForSmithyNamespace(smithyNamespace);
+      } catch (IllegalArgumentException e) {
+        // This is OK; just use the default module name
+        pythonModuleName = settings.getModuleName();
+      }
+      namespace = settings.getService().getNamespace();
+    } else {
+      pythonModuleName = getPythonModuleNameForSmithyNamespace(smithyNamespace);
+      namespace = smithyNamespace;
+    }
+    return pythonModuleName + ".smithygenerated." + getServiceSmithygeneratedDirectoryNameForNamespace(namespace);
+  }
+
+  /**
+   * Returns the name of the function that converts the provided shape's Dafny-modelled type to the
+   * corresponding Smithy-modelled type. This function will be defined in the `dafny_to_smithy.py`
+   * file. ex. example.namespace.ExampleShape -> "DafnyToSmithy_example_namespace_ExampleShape"
+   *
+   * This is the same as getSmithyToDafnyFunctionNameForShape below. These used to be different.
+   * There may be some value in preserving these separately if we want to make them different again in the future.
+   *
+   * @param shape
+   * @return
+   */
+  public static String getDafnyToSmithyFunctionNameForShape(
+      Shape shape, GenerationContext context) {
+    return SmithyNameResolver.getServiceSmithygeneratedDirectoryNameForNamespace(
+            shape.getId().getNamespace())
+        + "_"
+        + shape.getId().getName();
+  }
+
+  /**
+   * Returns the name of the function that converts the provided shape's Smithy-modelled type to the
+   * corresponding Dafny-modelled type. This function will be defined in the `smithy_to_dafny.py`
+   * file. ex. example.namespace.ExampleShape -> "SmithyToDafny_example_namespace_ExampleShape"
+   *
+   * This is the same as getDafnyToSmithyFunctionNameForShape above. These used to be different.
```

Comment:
I refactored into a shared implementation... was that what you were looking for?

Reply:
> These used to be different

It was a question that if they use the overloaded method now since they "used to be different". But I like the shared implementation.